src/shared.ts

```diff
@@ -115,7 +136,12 @@ export const loadScript = (
     }
   });
 
-  return stripePromise;
+  // set stripePromise to null on error
+  return stripePromise
+    .catch((error) => {
+      throw error;
+    })
+    .then((stripePromise = null));
```

Comment:
I recommend unsetting the `stripePromise` variable in the `catch` block. Putting it in a `then` block could cause the variable to be unset for a resolved promise.
```suggestion
  // Resets stripePromise on error
  return stripePromise
    .catch((error) => {
      stripePromise = null;
      return Promise.reject(error);
    });
```